cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -26,20 +26,32 @@ SNAP_UNIT_PREFIX="$(systemd-escape -p ${SNAP_MOUNT_DIR})"
 
 systemctl_stop() {
     unit="$1"
-    if systemctl is-active -q "$unit"; then
-        echo "Stoping $unit"
+    for i in $(seq 10); do
+        if systemctl is-active -q "$unit"; then
+            echo "Stoping $unit"
+            systemctl stop -q "$unit" || true
+        fi
+        echo "Stoping $unit [attempt $i]"
         systemctl stop -q "$unit" || true
-    fi
+        sleep .1
+    done
 }
 
 purge() {
-    # undo any bind mount to ${SNAP_MOUNT_DIR} that resulted from LP:#1668659
-    if grep -q "${SNAP_MOUNT_DIR} ${SNAP_MOUNT_DIR}" /proc/self/mountinfo; then
-        umount -l "${SNAP_MOUNT_DIR}" || true
+    if grep -q CODENAME=trusty /etc/os-release >/dev/null 2>&1 ; then
```

Comment:
Please don't do this. This is horribly brittle and broken. You have no idea what the content of this file is going to be, and you're hoping that this will work?

Also, you know `os-release(5)` is designed to be able to be sourced in shell, right? It works the same way as RH-style ifcfg files and any other key-value file.